libs/go-libs/sync/queue.go

```diff
@@ -0,0 +1,26 @@
+package sync
+
+type noCopy struct{}
+
+func (*noCopy) Lock()   {}
+func (*noCopy) Unlock() {}
+
+type Queue struct {
+	noCopy noCopy
```

Comment:
Why are you doing that? I'm curious

Reply:
Do you mean the Queue struct ? 
Or the noCopy ?